File: frontend/src/utils/jobDesignations.js
/** Job designation slugs and labels — keep in sync with backend/app/job_designations.py */

export const JOB_DESIGNATIONS = [
  { id: 'seo_expert', label: 'SEO Expert' },
  { id: 'smo_expert', label: 'SMO Expert' },
  { id: 'graphic_designer', label: 'Graphic Designer' },
  { id: 'digital_marketing_head', label: 'Digital Marketing Head' },
  { id: 'sales_executive', label: 'Sales Executive' },
  { id: 'sr_sales_executive', label: 'Sr. Sales Executive' },
  { id: 'sales_head', label: 'Sales Head' },
  { id: 'sales_director', label: 'Sales Director' },
  { id: 'operation_executive', label: 'Operation Executive' },
  { id: 'operation_head', label: 'Operation Head' },
  { id: 'operation_director', label: 'Operation Director' },
  { id: 'visa_filler', label: 'Visa Filler' },
  { id: 'tourist_visa_counsellor', label: 'Tourist Visa Counsellor' },
  { id: 'study_visa_counsellor', label: 'Study Visa Counsellor' },
  { id: 'education_loan_expert', label: 'Education Loan Expert' },
  { id: 'receptionist', label: 'Receptionist' },
  { id: 'accountant', label: 'Accountant' },
  { id: 'hr_assistant', label: 'HR Assistant' },
  { id: 'hr_head', label: 'HR Head' },
  { id: 'academic_counsellor', label: 'Academic Counsellor' },
];

export const LEGACY_JOB_TYPES = [
  { id: 'counselling', label: 'Counselling' },
  { id: 'sales', label: 'Sales' },
  { id: 'marketing', label: 'Marketing' },
  { id: 'operations', label: 'Operations' },
  { id: 'finance', label: 'Finance' },
  { id: 'hr', label: 'HR' },
  { id: 'it', label: 'IT & Technology' },
  { id: 'test_prep', label: 'Test Prep' },
  { id: 'visa', label: 'Visa & Documentation' },
  { id: 'management', label: 'Management' },
  { id: 'other', label: 'Other' },
];

export const JOB_TYPE_FILTERS = [
  { id: 'all', label: 'All Roles' },
  ...JOB_DESIGNATIONS,
  ...LEGACY_JOB_TYPES.filter(
    (legacy) => !JOB_DESIGNATIONS.some((d) => d.id === legacy.id),
  ),
];

export const TYPE_LABELS = Object.fromEntries(
  [...JOB_DESIGNATIONS, ...LEGACY_JOB_TYPES].map((t) => [t.id, t.label]),
);

export function jobTypeLabel(slug) {
  return TYPE_LABELS[slug] || slug.replace(/_/g, ' ').replace(/\b\w/g, (c) => c.toUpperCase());
}

File: backend/app/job_designations.py
"""Job designation slugs and labels for career postings."""

# Legacy department types (existing records may still use these)
LEGACY_JOB_TYPES: dict[str, str] = {
    "counselling": "Counselling",
    "sales": "Sales",
    "marketing": "Marketing",
    "operations": "Operations",
    "finance": "Finance",
    "hr": "HR",
    "it": "IT & Technology",
    "test_prep": "Test Prep",
    "visa": "Visa & Documentation",
    "management": "Management",
    "other": "Other",
}

# Vision International job designations
JOB_DESIGNATIONS: dict[str, str] = {
    "seo_expert": "SEO Expert",
    "smo_expert": "SMO Expert",
    "graphic_designer": "Graphic Designer",
    "digital_marketing_head": "Digital Marketing Head",
    "sales_executive": "Sales Executive",
    "sr_sales_executive": "Sr. Sales Executive",
    "sales_head": "Sales Head",
    "sales_director": "Sales Director",
    "operation_executive": "Operation Executive",
    "operation_head": "Operation Head",
    "operation_director": "Operation Director",
    "visa_filler": "Visa Filler",
    "tourist_visa_counsellor": "Tourist Visa Counsellor",
    "study_visa_counsellor": "Study Visa Counsellor",
    "education_loan_expert": "Education Loan Expert",
    "receptionist": "Receptionist",
    "accountant": "Accountant",
    "hr_assistant": "HR Assistant",
    "hr_head": "HR Head",
    "academic_counsellor": "Academic Counsellor",
}

ALL_JOB_TYPES: dict[str, str] = {**LEGACY_JOB_TYPES, **JOB_DESIGNATIONS}

JOB_TYPE_SLUGS: tuple[str, ...] = tuple(ALL_JOB_TYPES.keys())


def job_type_label(slug: str) -> str:
    return ALL_JOB_TYPES.get(slug, slug.replace("_", " ").title())


def is_valid_job_type(slug: str) -> bool:
    return slug in ALL_JOB_TYPES
